### DISCUSSION TITLE
Call API KO ?

### DISCUSSION THREAD

**Discussion by Hugz59** [osm-search/Nominatim#2844] [Q&A]
Hello everyone ! 

I would like to come to you because I am very confused. 

I try to call the https://nominatim.openstreetmap.org/search?<params> API from a PHP file but it doesn’t work. 

Indeed: 

If I put the following url in my search bar, I get the json file: https://nominatim.openstreetmap.org/search.php?q=135+pilkington+avenue%2C+birmingham&format=jsonv2

If I call it in my file I have an error 404 not found and via a curl -v I have a 302: 

`
Location: https://nominatim.openstreetmap.org/ui/search.html?q=135+pilkington+avenue%2C+birmingham`

If I try to put https://nominatim.openstreetmap.org/ui/search.html?q=135+pilkington+avenue%2C+birmingham&format=json in my file, I get a blank page and cannot retrieve the parameters. 

There is my php file : 

```
<?php

$curl= curl_init('https://nominatim.openstreetmap.org/ui/search.html?q=135+pilkington+avenue%2C+birmingham&format=json');
curl_setopt($curl, CURLOPT_CAINFO, 'D:\wamp\www\\cacert.pem');
$data = curl_exec($curl); 

if($data === false){
    var_dump(curl_error($curl));
}
else {

}

curl_close($curl);

?>
```

How can i do for get the json ?

**Comment by mtmail** [comment]
/ui/search.html is not the correct API Url, it should be https://nominatim.openstreetmap.org/search

You should add a user agent string identifying your application 
https://operations.osmfoundation.org/policies/nominatim/ Afaik the default curl user agent is blocked.

**Reply by Hugz59** [reply]
Oh thanks it's working ! Thank you very much !
